### ISSUE TITLE
Provenance related display problems

### ISSUE THREAD

**Issue by gothub** [NCEAS/metacatui#81]
For the package https://dev.nceas.ucsb.edu/#view/doi:10.5063/F1KD1VVS, there are the following provenance related issues:

- the amount of display real estate is significantly reduced by the provenance related icons. See the data entity 'CO_spills_final_dataviz.csv'. The 'Data Table' for this entity is about as wide as the prov icons that are on both sides - two  column for future edit icons, two column for programs, two columns for derivations or sources. Can we make the icons smaller maybe?
- for this same entity, there is a gap between the '2 sources' and the data table, as there is no program involved (sources are related to the entity by [prov_wasDerivedFrom relationships](https://dev.nceas.ucsb.edu/knb/d1/mn/v1/query/solr/?q=id:%22brun.39.1%22)).
- for entity 'US States', the 'derivations' icon list is incomplete - the text says '10 derivations', but only 6 are displayed
- the text for derivations for the the next entity  'Shale plays dissolved' is vertically compressed
- the 'inputs' icons for 'DataViz_Analysis.R' and all following entities are only one color

**Comment by amoeba** [comment]
Also just ran into this (minor) issue: The prov UI is rendered at a different zindex than the metadata content:
![screencast](https://cloud.githubusercontent.com/assets/563/25020506/199f05ee-203b-11e7-8937-dd629841f7d9.gif)

**Comment by rushirajnenuji** [comment]
There is issue regarding display with different browser widths:

![prov 1](https://user-images.githubusercontent.com/19696935/32861973-60983ae0-ca0b-11e7-8424-a7eaa5d3e4c6.png)

![prov 2](https://user-images.githubusercontent.com/19696935/32861978-63fb8af2-ca0b-11e7-95ba-c3969f00e32a.png)

**Comment by laurenwalker** [comment]
I fixed all these issues except for "the 'inputs' icons for 'DataViz_Analysis.R' and all following entities are only one color" - This is something that requires changes to the ProvChartView rendering process. 

Right now we assign colors to the nodes by adding unique class names and applying colors to those classes via CSS. After the class names run out, we assign the same color to all nodes. This was based on the idea that there are somewhere between 24 different colors that people can easily distinguish in a chart or graph. But with large data packages, we have a need to apply hundreds of  different colors, even if they won't be easily distinguishable.

I made a new ticket for this issue: #407
